network.go

```diff
@@ -53,73 +53,66 @@ func linkByHwAddr(netHandle *netlink.Handle, hwAddr string) (netlink.Link, error
 	return nil, fmt.Errorf("Could not find the link corresponding to HwAddr %q", hwAddr)
 }
 
-func updateInterfaceAddrs(netHandle *netlink.Handle, link netlink.Link, addrs []*pb.IPAddress, add bool) error {
-	for _, addr := range addrs {
-		netlinkAddrStr := fmt.Sprintf("%s/%s", addr.Address, addr.Mask)
+func updateLink(netHandle *netlink.Handle, link netlink.Link, iface *pb.Interface) error {
+
+	// As a first step, clear out any existing addresses associated with the link:
```

Comment:
With clear and re-add we will lose any routes related to those addresses(e.g. default route via a gateway address in the same subnet). IMHO, we should compute diff of addresses and apply it instead.